# Homework 1 

## AMPTH 207: Stochastic Methods for Data Analysis, Inference and Optimization

**Due Date:** Thursday, Febrary 2nd, 2017 at 11:59pm

**Instructions:**

- Upload your final answers as well as your iPython notebook containing all work to Canvas.

- Structure your notebook and your work to maximize readability.

## Problem 1: Bayes Theorem

**Coding not required**

### Part A:
Your child has been randomly selected for Type I diabetes screening, using a highly accurate new test that boasts of a false positive rate of 1% and a false negative rate of 0%. The prevalence of of Type I diabetes in children is approximately 0.228%. Should your child test positive, what is the probability that he/she has Type I diabetes?

Should you be concerned enough to ask for further testing or treatment for your child?

Later, you read online that Type I diabetes is 6 times more prevalent in prematurely born children. If this statistic is true, what is the probability that your child, who is prematurely born, has Type I diabetes?

Given the new information, should you be concerned enough to ask for further testing or treatment for your child?

**Justify your decisions using your calculations.**

### Answer:

Given Bayes' Theorem: $$ p(y\mid x) = \frac{p(x\mid y) \, p(y) }{p(x)} = \frac{p(x\mid y) \, p(y) }{\sum_{y'} p(x,y')} = \frac{p(x\mid y) \, p(y) }{\sum_{y'} p(x\mid y')p(y')}$$

We adapt this to the probabilities of having Type I diabetes (D), not having type I diabetes aka healthy (H), testing positive (Y), and testing negative (N). These are given above and defined in code below.

In [10]:
D = 0.00228
H = 1 - D

YH = 0.01 # P(Y|H), aka false positive rate
ND = 0 # P(N|D), aka false negative rate
YD = 1 - ND # P(Y|D), aka true positive; inverse of false negative
NH = 1 - YH # P(N|H), aka true negative; inverse of false positive. Not used below.

To calculate the probability of my child having Type I diabetes given a positive test:

$$ p(D\mid Y) = \frac{p(Y\mid D) \, p(D) }{p(Y)} = \frac{p(Y\mid D) \, p(D) }{\sum_{D'} p(Y\mid D')p(D')}$$

$$ = \frac{p(Y\mid D) \, p(D) }{p(Y\mid D) p(D) + p(Y\mid H) p(H)}$$

Let's calculate it using the values defined above.

In [7]:
DY = (YD*D)/((YD*D)+(YH*H))
DY # Probability of having disease given a positive test

0.18601311881995888

#### **Probability of child actually having Type I diabetes is 18.6%.** 

This is fairly low, but enough that I would probably ask for more testing. 

Now, upon the discovery that Type I diabetes is 6 times more prevalent in premature babies, this changes $D$ to $0.00228*6$, so it is now $D = 0.01368$. Correspondingly, we must update $H$ to $1 - D$.

In [9]:
D = 0.01368
H = 1 - D

DYpremature = (YD*D)/((YD*D)+(YH*H))
DYpremature # Probability of having disease given a positive test, if a premature baby

0.5810594991335077

#### **There is a 58.1% chance that the positive test is correct.**

At this point I would absolutely ask for further testing or treatment, given the severe complications that can arise from untreated Type I diabetes.

### Part B:

During shopping week, you're trying to decide between two classes based on the criteria that the class must have a lenient grading system. You hear from your friends that one of these classes is rumored to award grades lower than the work merits 35% of the time while the other awards lower grades 15% of the time. However, the rumor doesn't specify which class has harsher grading. So, you decide to conduct an experiment: submit an assignment to be graded. 

Fortunately, both classes offer an optional Homework 0 that is graded as extra credit. Unfortunately, you only have time to complete the problem set for just one of these classes. 

Suppose you randomly pick the Homework 0 from Class A to complete and suppose that you received a grade that you believe is lower than the quality of your work warrents. Based on this evidence, what is the probability that Class A has the harsher grading system? Which class should you drop based on the results of your experiment (or do you not have sufficient evidence to decide)?

**Justify your decisions using your calculations.**

### Answer:

Let's assign our mystery classes the designations Class Hard (15%  of the time, the grade is lower) and Class Harder (35% of the time, the grade is lower). 

The probability of getting a low grade is therefore as follows:

$p(low \mid hard) = 0.15$

$p(low \mid harder) = 0.35$

We know that there are two classes and we've chosen one by random, so:

$p(hard) = 0.5$ 

$p(harder) = 0.5$.

To calculate the probability of Class A being harder, given what we believe to be an unfairly low grade on Homework 0, we again use Bayes' Theorem. 

$$ p(harder\mid low) = \frac{p(low\mid harder) \, p(harder) }{p(low\mid harder) p(harder) + p(low\mid hard) p(hard)}$$

In [12]:
LHard = 0.15
LHarder = 0.35
Hard = 0.5
Harder = 0.5

isHarsher = (LHarder*Harder)/((LHarder*Harder)+(LHard*Hard))
isHarsher # What is the probability that Class A is the harsher grading system?

0.7

#### **There is a 70% chance that Class A has the harsher grading system.** 

This assumes I have correctly determined that my work deserves a higher grade (which is debatable). This also assumes that the the magnitude of the unwarranted score drop is similar between the two courses. All else being equal, if I were determined to drop one of these courses, **I would drop Course A.** 

## Problem 2: Maximum Likelihood Estimates

**Coding required for Part B only**

Suppose you observe the following data set $\mathbf{x}^{(0)} = (0.5, 2.5), \mathbf{x}^{(1)} = (3.2, 1.3), \mathbf{x}^{(2)} = (2.72, 5.84), \mathbf{x}^{(3)}= (10.047, 0.354)$. By convention, for any vector $\mathbf{x}$, we will denote the first component of $\mathbf{x}$ by $x_{1}$ and the second component by $x_{2}$. Suppose that the data is drawn from the same two-dimensional probability distribution with pdf $f_X$, that is, $\mathbf{x}^{(i)} \overset{iid}{\sim} f_X$, where
$$
f_X(\mathbf{x}) =  4\lambda_1^2 x_{1}x_{2} \mathrm{exp} \left\{-\lambda_0 (x^2_{1} + x^2_{2}) \right\}.
$$
You should assume that $\lambda_1, \lambda_0 > 0$ and that $f_X$ is supported on the nonnegative quandrant of $\mathbb{R}^2$ (i.e. $f_X$ is zero when either component is negative). 

### Part A:

What are the values for $\lambda_0$ and $\lambda_1$ that maximizes the likelihood of the observed data? 

**Support your answer with full and rigorous analytic derivations.**

### Part B:

Visualize the data along with the distribution you determined in part A (in two dimensions or three).

## Problem 3: Frequentist Stats
**Coding required**

Read the data set contained in [Homework_1_Data.txt](Homework_1_Data.txt). Each data point is a two-dimensional vector, $\mathbf{x} = (x_1, x_2)$.

### Part (A): Visualization and Interpretation

- Make a 2-D visualization the distribution of the data. 

- Visualize the the pdf, $f_X$, of the underlying distribution of the data.

- Visualize the distribution defined by $f_{x_2 \mid x_1}$ for $x_1 \in [3.99, 4.01]$.

- Visualize the distribution defined by $f_{x_1}$.


### Part (B): Estimation

- Empirically estimate the mean of the distribution $f_{x_1}$. Estimate, also the SE (standard error) of the estimate.

- Empirically estimate the standard deviation of the distribution $f_{x_2 \mid x_1}$, for $x_1 \in [3.99, 4.01]$. Estimate, also the SE (standard error) of the estimate.

- Given the SE, How many digits in your standard deviation estimate are significant? Explain why.